### ISSUE TITLE
main extension default-action shortcut bug 

### ISSUE THREAD

**Issue by gittyup2018** [aclap-dev/vdhcoapp#208]
Hi, 

I couldn't find a issues section on the main extension's repo so I'm posting the bug (in Firefox) I found here: 

The Execute default action shortcut doesn't seem to recognize when a different action (like copy url) is selected and will download the stream instead of using the appropriate action (like copy url) 

I have the default action of Copy Url, and that copies the URL when I click on an item from the popup correctly, but when using the shortcut the extension tries to download the stream instead of copying it. I'm sure it's just a little thing like a typo or something easily fixable but I can't find your souce code's repo so I don't think it's open source so can't fix it myself..

**Comment by paulrouget** [comment]
Thank you for the bug report. Please add more information about your OS/Browser/Version: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

**Comment by gittyup2018** [comment]
Same issue on both Firefox and Chrome..the default-action is never changed. I didn't test this on Windows but imagine it uses the same JavaScript code so it's the same..

Firefox>

version: 8.1.4.0
beta: false
build date: 2024-01-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0

Chrome>

version: 8.1.6.0
beta: false
build date: 2024-01-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36

**Comment by paulrouget** [comment]
Do you mind testing with v8.2? https://downloadhelper.net/beta

**Comment by gittyup2018** [comment]
I tested the betas in both FF & Chrome and now absolutely nothing happens. No popup, no error and not even a console message like the shortcut key was never even pressed.

**Comment by paulrouget** [comment]
Understood! I'll look into it. Thank you.

**Comment by paulrouget** [comment]
I fixed it in v8.2.0.21. It will available in the google/mozilla/microsoft addon website in a few hours, and it will autoupdate in 24 hours.

**Comment by paulrouget** [comment]
See https://github.com/aclap-dev/video-downloadhelper/discussions/82

**Comment by gittyup2018** [comment]
Thanks that works! Is there a way to use the "old" download detector? on testing several sites it doesn't copy the URL correctly. I tried the exact same sites with the beta version and regular and regular worked fine. So I really hope there is just a switch to disable the new version?

**Comment by paulrouget** [comment]
> on testing several sites it doesn't copy the URL correctly

The copy url changed, indeed. You're not the first one to complain about this. You can see in the details, there's 2 (or 3) URLs. We were copying the media url before, now we copy the manifest url.

You prefer the old behavior?

I might backtrack on this, and use the old URL.

**Comment by gittyup2018** [comment]
Yes I never had a issue with this extension before..the only issue I ever had is that default-action which you fixed :) but yes please make it a setting or something at the very least

**Comment by gittyup2018** [comment]
To go 1 step further, if it was a setting (in a very future version) then you can  make it site-specific to choose which detect method works best so can use media url on site A and manifest url on site B, etc.

**Comment by paulrouget** [comment]
Ok, so, in the upcoming 8.2.0.24 (wait a day or two before it's available), we reverted to media manifest instead of master manifest.

**Comment by gittyup2018** [comment]
Cool looking forward to it! I'll test it and let you know if there's any issues. I'm still using 8.1 so just wondering is the 8.2.0.24 a beta version or regular?

**Comment by gittyup2018** [comment]
Is 8.0.24 a typo? I'm on 8.0.23 and that seems to be the newest. This is starting to get very annoying. If your code was open source I could have fixed it myself and tested it, which obviously you never did when you implement all these features. I imagine this copy URL bug existed since version 1.0 but no one ever tested it, including yourself  which is mind blowing and makes me think you must be self-taught.

**Comment by paulrouget** [comment]
@gittyup2018 we do not engage with rude people. Have a good day.
